cache/refs.go

```diff
@@ -32,33 +30,19 @@ import (
 // Ref is a reference to cacheable objects.
 type Ref interface {
 	Mountable
-	ID() string
+	Metadata
```

Comment:
No change needed, but I kind of liked it as separate from the rest of the implementation. Guess it can still be nullable if implementation just defines embedded interface as well.

Reply:
I just found that having all the getters/setters as methods on refs rather than detached functions made it way easier to write code and remember what metadata exists. It works much better with IDE autocompletion and is slightly less verbose.